lib/live_view_native/swiftui/styles/utility.ex

```diff
@@ -119,40 +120,48 @@ defmodule LiveViewNative.SwiftUI.UtilityStyles do
   end
 
   for modifier <- Enum.uniq(@modifier_names) |> Enum.sort_by(&String.length/1, :desc) do
-    kebab_name =
-      modifier
-      |> Macro.underscore()
-      |> String.replace("_", "-")
-
-    def class(unquote(kebab_name) <> arguments) do
+    def class(unquote(modifier) <> arguments = class_name) do
       name = unquote(modifier)
 
-      arguments =
-        arguments
-        |> String.trim_leading("-") # remove dash separating first argument
-        |> String.split(~r/(?<!-|:)-/) # arguments separated by a dash
-        |> Enum.map(fn arg ->
-          if String.contains?(arg, ":") and not String.starts_with?(arg, ":") do
-            [name | value] = String.split(arg, ":")
-            value = Enum.join(value, ":")
-            "#{name}: #{encode_argument(value)}" # add space between label and value
-          else
-            "#{encode_argument(arg)}" # encode argument values
-          end
-        end)
-        |> Enum.join(", ") # rejoin arguments with commas instead of dashes
-        |> String.replace(":.", ": .")
       try do
+        arguments = LiveViewNative.SwiftUI.UtilityStyles.parse_arguments(arguments)
         ~RULES"""
         <%= name %>(<%= arguments %>)
         """
       rescue
         _ ->
-          {:unmatched, ""}
+          {:unmatched, "Stylesheet warning: Could not match on class: #{class_name}"}
       end
     end
   end
 
+  def parse_arguments(<<?(, arguments::binary>>) do
+    <<?), arguments::binary>> = String.reverse(arguments)
+
+    arguments
+    |> String.reverse()
+    |> String.replace(":", ": ")
+    |> String.replace(",", ", ")
```

Comment:
Would this mess up string arguments that contain `:` or `,`?

It would be nice if the rules parser didn't require the whitespace and we could send it straight through.

Reply:
yeah that's a good point

@NduatiK can we support both whitespace and non-whitespacing in the rules parser?

For example:

`foo(bar:1,bar:2)` and `foo(bar: 1, bar: 2)`

Reply:
Sure thing.

I'm pretty backed up at work so earliest I can get to this is over the weekend.